packages/lib/isOutOfBounds.tsx

```diff
@@ -226,6 +227,37 @@ export function isTimeOutOfBounds({
   return false;
 }
 
+/**
+ * Wrapper over isTimeOutOfBounds to return a status object.
+ * Note: It doesn't throw any error and can be safely used
+ */
+export function getTimeOutOfBoundsStatus({
+  time,
+  minimumBookingNotice,
+}: {
+  time: dayjs.ConfigType;
+  minimumBookingNotice?: number;
+}): {
+  isOutOfBounds: boolean;
+  reason: "minBookNoticeViolation" | "slotInPast" | null;
+} {
+  try {
+    const isOutOfBounds = isTimeOutOfBounds({ time, minimumBookingNotice });
+    return {
+      isOutOfBounds,
+      reason: isOutOfBounds ? "minBookNoticeViolation" : null,
+    };
+  } catch (error) {
+    if (error instanceof BookingDateInPastError) {
+      return {
+        isOutOfBounds: true,
+        reason: "slotInPast",
+      };
+    }
+    throw error;
+  }
+}
```

Comment:
Technically, minimumBookingNotice is only one of the reasons why we might face`booking_time_out_of_bounds_error`, so for clarity purposes I'd recommend we either rename this check or include check for the other factors that might lead to the said error.

The other factors are when the user has “limit future bookings” selected,
- Rolling: This would fail if the booking slot selected is far into the future (beyond 30 days, for example when 30 days is selected here)
- Rolling-Window: This would fail if the booking slot selected is far into the future (beyond n days of “always ‘n’ days available”)
- Range: This would fail if the booking slot selected is outside of the chosen range

Reply:
Renamed